src/index.ts

```diff
@@ -1,139 +1,115 @@
-            }
-            break;
-
-        case 'theme':
-            switch (parseArgs._[1]) {
-                case 'serve':
-                    await createDevelopmentServer(
-                        join(process.cwd(), parseArgs[Argument.ThemePath]),
-                        parseArgs[Argument.EntryPath],
-                        parseArgs[Argument.Port]
-                    );
-                    break;
-                case 'deploy':
-                    await createDeployment(
-                        parseArgs[Argument.ThemePath],
-                        parseArgs[Argument.EntryPath],
-                        parseArgs[Argument.OutDir],
-                        {
-                            dryRun: parseArgs[Argument.DryRun],
-                            noVerify: parseArgs[Argument.NoVerify],
-                            openInBrowser: parseArgs.open,
-                        }
-                    );
-                    break;
+import { createDeployment, createDevelopmentServer, createNewContentBlock, loginUser, logoutUser } from './commands';
+import { getValidInstanceUrl } from './utils';
+import pkg from '../package.json';
+
+const cli = cac(pkg.name.split('/')[1]);
+
+cli.command('login [instanceUrl]', 'log in to a Frontify instance')
+    .option('-i, --instance [instanceUrl]', '[string] url of the Frontify instance')
+    .option('-p, --port <port>', '[number] port for the oauth service', {
+        default: process.env.PORT || 5600,
+    })
+    .action(async (instanceUrl: string, options) => {
+        let promptedInstanceUrl: string | null = null;
+
+        if (!instanceUrl && !options.instance && !process.env.INSTANCE_URL) {
+            const { value } = await prompts({
+                type: 'text',
+                name: 'value',
+                message: 'Frontify Instance URL',
+                initial: 'instanceName.frontify.com',
+                validate: (value: string) => (value === '' ? 'You need to enter a URL.' : true),
+            });
+
+            if (!value) {
+                exit(0);
             }
-            break;
 
-        case 'login':
-            const instanceUrl = getValidInstanceUrl(parseArgs.instance || process.env.INSTANCE_URL);
-            await loginUser(instanceUrl, parseArgs[Argument.Port]);
-            break;
+            promptedInstanceUrl = value;
+        }
+
+        const parsedInstanceUrl = getValidInstanceUrl(
+            instanceUrl ?? options.instance ?? process.env.INSTANCE_URL ?? promptedInstanceUrl
+        );
+
+        await loginUser(parsedInstanceUrl, options.port);
+    });
+
+cli.command('logout', 'log out of an instance').action(logoutUser);
 
-        case 'logout':
-            logoutUser();
-            exit(1);
+/**
+ * @deprecated `block serve` and `theme serve` will be removed in version 4.0 in favour of `serve`
+ */
+for (const appType of ['block', 'theme']) {
+    cli.command(`[root] ${appType} serve`, `serve the ${appType} locally`)
```

Comment:
Maybe mention it here as well that it's being deprecated in v4.0, so the user can immediately see it?

```suggestion
    cli.command(`[root] ${appType} serve`, `[deprecated: use 'serve' instead] serve the ${appType} locally`)
```